File: 20190519python/0204. Count Primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n==0 or n==1:
            return 0
        listn=[0]*n
        listn[0]=1
        listn[1]=1
        for i in range(2,n//2+1):
            if listn[i]==0:
                k=2
                while i*k<n:
                    listn[i*k]=1
                    k+=1
        rep=0
        for num in listn:
            if num==0:
                rep+=1
        return rep

File: archives/C#/0204. Count Primes.cs
public class Solution {
    public int CountPrimes(int n) {
        if(n==0 || n==1){
            return 0;
        }
        int[] nums=new int[n];
        nums[0]=1;
        nums[1]=1;
        for(int i=2;i<=(n-1)/2;i++){
            if(nums[i]==0){
                int j=2;
                while(i*j<n){
                    nums[i*j]=1;
                    j+=1;
                }
            }
        }
        int rep=0;
        foreach(int num in nums){
            if (num==0){
                rep+=1;
            }
        }
        return rep;
    }
}
